### DISCUSSION TITLE
GWM ORA battery health TX, RX and expression

### DISCUSSION THREAD

**Discussion by holger1414** [meatpiHQ/wican-fw#156] [Q&A] [answered]
Hi!

Do you happen to have any info on that sensor?
Thank you very much for your project!

Holger

**Comment by meatpiHQ** [comment]
Hey @holger1414 Please try the new beta version, this will make it easier for you to get the battery SoC. 

With this firmware you can setup WiCAN to publish the SoC without the TX frame.  [wican-fw_v310beta_hv300.zip](https://github.com/user-attachments/files/16160263/wican-fw_v310beta_hv300.zip)

Here how to set it up, Frist you need to update the device, just extract the bin file and [update](https://github.com/meatpiHQ/wican-fw?tab=readme-ov-file#1-ota)

Go to Automate tab, and fill in these parameters, click the "**store**" button then reboot the device. after reboot when it connects it will start publishing the soc to topic: log_soc

**Name**: SoC
**PID**: 220308
**Expression**: [B4:B5]/10
**Period**: 1000
**Send_to**: log_soc

<img width="881" alt="image" src="https://github.com/meatpiHQ/wican-fw/assets/94690098/cddf9f6e-d5c3-4618-97f5-ab711307618c">

**Comment by holger1414** [comment]
Hi! Thanks for the quick response!
However, I am asking about battery HEALTH :)

I have SoC and mileage working very well.

**Reply by meatpiHQ** [reply]
Sorry I assumed you wanted the SoC. 

Does carscanner or any other app read the health?

**Reply by holger1414** [reply]
My bad!
I was replying via email and included screenshot, but it did not make it to discussion :)

{"bus":"0","type":"rx","ts":2305,"frame":[{"id":1995,"dlc":8,"rtr":false,"extd":false,"data":[5,98,32,81,3,165,204,204]}]}
Car scanner value 93,3%

This value has not changed yet, so I can not provide additional data :/

![image](https://github.com/meatpiHQ/wican-fw/assets/82466516/ff2770a7-c52c-4988-903d-3e5e2f63dedd)

**Reply by meatpiHQ** [reply, marked as answer]
what is the TX frame?

This expression should work, ((B4*256)+B5)/10

**Reply by holger1414** [reply]
Amazing!
It worked!
So the expression for SoC and Healt are the same. Was not expecting that :)

Thank you!!

BTW TX frame is
{ "bus": "0", "type": "tx", "ts": 18461, "frame": [{ "id": 1931, "dlc": 8, "rtr": false, "extd": false, "data": [3, 34, 32, 81, 170, 170, 170, 170]}]}
